ambry-cloud/src/test/java/com.github.ambry.cloud/CloudBlobStoreTest.java

```diff
@@ -614,4 +615,120 @@ private BlobId getUniqueId(short accountId, short containerId, boolean encrypted
+        new LatchBasedInMemoryCloudDestination(blobIdList);
+    CloudReplica cloudReplica = new CloudReplica(cloudConfig, partitionId, cloudDataNode);
+    CloudBlobStore cloudBlobStore =
+        new CloudBlobStore(new VerifiableProperties(props), partitionId, latchBasedInMemoryCloudDestination, clusterMap,
+            new VcrMetrics(new MetricRegistry()));
+    cloudBlobStore.start();
+
+    // Create ReplicaThread and add RemoteReplicaInfo to it.
+    ReplicationMetrics replicationMetrics = new ReplicationMetrics(new MetricRegistry(), Collections.emptyList());
+    ReplicaThread replicaThread =
+        new ReplicaThread("threadtest", new MockFindToken.MockFindTokenFactory(), clusterMap, new AtomicInteger(0),
+            cloudDataNode, connectionPool, replicationConfig, replicationMetrics, null, storeKeyConverter, transformer,
+            clusterMap.getMetricRegistry(), false, cloudDataNode.getDatacenterName(), new ResponseHandler(clusterMap),
+            new MockTime());
+
+    for (ReplicaId replica : partitionId.getReplicaIds()) {
+      if (replica.getDataNodeId() == remoteHost.dataNodeId) {
+        RemoteReplicaInfo remoteReplicaInfo =
+            new RemoteReplicaInfo(replica, cloudReplica, cloudBlobStore, new MockFindToken(0, 0), Long.MAX_VALUE,
+                SystemTime.getInstance(), new Port(remoteHost.dataNodeId.getPort(), PortType.PLAINTEXT));
+        replicaThread.addRemoteReplicaInfo(remoteReplicaInfo);
+        break;
+      }
+    }
+
+    long referenceTime = System.currentTimeMillis();
+    BlobId id = blobIdList.get(0);
+    addPutMessagesToReplicasOfPartition(id, accountId, containerId, partitionId, Collections.singletonList(remoteHost),
+        referenceTime - 2000, referenceTime - 1000);
+    addTtlUpdateMessagesToReplicasOfPartition(partitionId, id, Collections.singletonList(remoteHost),
+        Utils.Infinite_Time);
+    id = blobIdList.get(1);
+    addPutMessagesToReplicasOfPartition(id, accountId, containerId, partitionId, Collections.singletonList(remoteHost),
+        referenceTime - 2000, referenceTime - 1000);
+    addTtlUpdateMessagesToReplicasOfPartition(partitionId, id, Collections.singletonList(remoteHost),
+        Utils.Infinite_Time);
+    replicaThread.replicate();
+    assertTrue("Blob should exist.", latchBasedInMemoryCloudDestination.doesBlobExist(id));
+
+    //try get for a blob that exists
+    List<BlobId> blobIds = new ArrayList<>(1);
+    blobIds.add(blobIdList.get(0));
+    StoreInfo storeInfo = cloudBlobStore.get(blobIds, null);
+    assert (storeInfo.getMessageReadSetInfo().get(0).getStoreKey().equals(blobIdList.get(0)));
+
+    //try get for a list of blobs that exist
+    blobIds = new ArrayList<>(2);
+    blobIds.add(blobIdList.get(0));
+    blobIds.add(blobIdList.get(1));
+    storeInfo = cloudBlobStore.get(blobIds, null);
+    assert (storeInfo.getMessageReadSetInfo().get(0).getStoreKey().equals(blobIdList.get(0)));
```

Comment:
We usually use `org.junit.Assert.*`. You can use `assertTrue` here.

Reply:
sure.